### PR TITLE
Adds Timeout support to RequestFactories Http implementations.

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR adds Timeout support to RequestFactories Http implementations.

Also, adds the default timeout for `CIVisibility` mode.

### CODE REVIEW

**Review by lucaspimentel — APPROVED** [review]

File: tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs [hunk 1]
```diff
@@ -77,12 +77,14 @@ protected override IAgentWriter GetAgentWriter(ImmutableTracerSettings settings,
         private IApiRequestFactory GetRequestFactory()
         {
             IApiRequestFactory factory = null;
+            TimeSpan agentlessTimeout = TimeSpan.FromSeconds(20);
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs:80]: 20s still seems quite long to me? I was thinking more like 5s, especially with the retries? IMO, a smaller timeout, and larger delay between retries may be preferable, but not passionate about it

**Review by andrewlock — APPROVED** [review]

File: tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs (hunk 1 above)
Comment (tonyredondo) [reply tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs:80]: yes, we decide 15secs, but the Timeout in .NET includes the DNS resolver, and that can take 15secs 🤦🏻 . So I need to go above of that number.
https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-6.0#remarks

Comment (tonyredondo) [reply tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs:80]: oh wait, is separated, not included. Will reduce it to 15secs.

Comment (tonyredondo) [reply tracer/src/Datadog.Trace/Ci/CITracerManagerFactory.cs]: done in [667db3a](https://github.com/DataDog/dd-trace-dotnet/pull/2548/commits/667db3a76d26152885c3807a251572ea737c9c9f)

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #2548 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6f1635f2181159e96c2a83c1c217440b08ba8add) will **not change** line coverage
:heavy_check_mark: Merging #2548 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6f1635f2181159e96c2a83c1c217440b08ba8add) will **not change** branch coverage
:no_entry: Merging #2548 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/6f1635f2181159e96c2a83c1c217440b08ba8add) will will **increase** complexity by `4`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/6f1635f2181159e96c2a83c1c217440b08ba8add) | #2548       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `11959` / `16852` | `11958` / `16858` |          |
| Lines %   | `71%`      | `71%`      |  `0%` :heavy_check_mark:  |
| Branches  | `6467` / `9788` | `6462` / `9792` |          |
| Branches %| `66%`      | `66%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `11056`      | `11060`        |  `4`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=92989&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/92989/artifacts?artifactName=Code%20Coverage%20Report_92989&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2548](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93076&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93076/artifacts?artifactName=Code%20Coverage%20Report_93076&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/6f1635f2181159e96c2a83c1c217440b08ba8add) | #2548       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `71%`       |  `0%` :heavy_check_mark:  |
| Branches %| `66%`      | `66%`       |  `0%` :heavy_check_mark:  |
| Complexity| `11056`      | `11060`       |  `4` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Telemetry.JsonWebRequestTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/6ce29365975c7ef1237c4efafe36ad4db6875f5d/tracer/src/Datadog.Trace/Telemetry/Transports/JsonWebRequestTelemetryTransport.cs) | `-28%` :no_entry: | `-57%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ClrProfiler.AutoInstrumentation.MongoDb.MongoDbIntegration](https://github.com/DataDog/dd-trace-dotnet/tree/6ce29365975c7ef1237c4efafe36ad4db6875f5d/tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbIntegration.cs) | `-7%` :no_entry: | `0%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Telemetry.JsonHttpClientTelemetryTransport](https://github.com/DataDog/dd-trace-dotnet/tree/6ce29365975c7ef1237c4efafe36ad4db6875f5d/tracer/src/Datadog.Trace/Telemetry/Transports/JsonHttpClientTelemetryTransport.cs) | `26%` :heavy_check_mark: | `38%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=92989&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/92989/artifacts?artifactName=Code%20Coverage%20Report_92989&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2548](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=93076&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/93076/artifacts?artifactName=Code%20Coverage%20Report_93076&api-version=6.0&%24format=zip)

**Pull request merged**